pkg/pipeline/ingest/ingest_kafka.go

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) 2022 IBM, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *	 http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package ingest
+
+import (
+	"encoding/json"
+	"errors"
+	"github.com/netobserv/flowlogs2metrics/pkg/api"
+	"github.com/netobserv/flowlogs2metrics/pkg/config"
+	kafka "github.com/segmentio/kafka-go"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/net/context"
+)
+
+type ingestKafka struct {
+	kafkaParams api.IngestKafka
+	kafkaReader *kafka.Reader
+}
+
+// Ingest ingests entries from kafka topic and sends them down the pipeline
+func (r *ingestKafka) Ingest(process ProcessFunction) {
+	lines := make([]interface{}, 1)
+
+	for {
+		m, err := r.kafkaReader.ReadMessage(context.Background())
+		if err != nil {
+			log.Errorln(err)
+		}
+		log.Debugf("message at topic:%v partition:%v offset:%v	%s = %s\n", m.Topic, m.Partition, m.Offset, string(m.Key), string(m.Value))
+		lines[0] = string(m.Value)
```

Comment:
So, are we getting here one line by the other, or bulk of messages ... why this is like lines[0] ... this will make the "process" function much less efficient

Reply:
We read whatever the kafka topic has to offer. I did not do here any parsing of the input, since I don't yet know what to expect. The rest of the pipeline expects an array of lines. If we know the kinds of divisions expected in the kafka, then I should divide it up here into "lines".

Reply:
@KalmanMeth ok ... so this is maybe another good reason to have some simple test to show what kind of input we expect to get in kafaka. 

@mariomac @jpinsonneau any data to share on the expected format of Kafka messages?

Reply:
Each message is the output of flow collector before or after enrichment and other format processing from the pipeline so I guess there is no expected format other than a valid `JSON` ?

Here is an example of a single message we get after enrichment:
`{"BiFlowDirection":0,"Bytes":20800,"DstAS":0,"DstAddr":"10.96.0.1","DstMac":"0a:58:0a:f4:00:01","DstNet":0,"DstPort":443,"DstVlan":0,"EgressVrfID":0,"Etype":2048,"EtypeName":"IPv4","ForwardingStatus":0,"FragmentId":0,"FragmentOffset":0,"IPTTL":0,"IPTos":0,"IPv6FlowLabel":0,"IcmpCode":0,"IcmpName":"","IcmpType":0,"InIf":12,"IngressVrfID":0,"NextHop":"","NextHopAS":0,"OutIf":0,"Packets":400,"Proto":6,"ProtoName":"TCP","SamplerAddress":"10.244.0.2","SamplingRate":0,"SequenceNum":577,"SrcAS":0,"SrcAddr":"10.244.0.5","SrcHostIP":"10.89.0.2","SrcMac":"0a:58:0a:f4:00:05","SrcNamespace":"local-path-storage","SrcNet":0,"SrcPod":"local-path-provisioner-78776bfc44-p2xkl","SrcPort":56144,"SrcVlan":0,"SrcWorkload":"local-path-provisioner","SrcWorkloadKind":"Deployment","TCPFlags":0,"TimeFlowEnd":0,"TimeFlowStart":0,"TimeReceived":1628419398,"Type":"IPFIX","VlanId":0}`

For now on our side, if some mandatory fields are missing, the messages will be filtered by the queries on loki side and not displayed in the console plugin. 
Do you think we should have an option here to check some user specified fields / values before writing to kafka and skip messages that doesn't match ?

Reply:
@jpinsonneau thanks :-) .... what you shared is the raw json after enrichment ... do you have some specific Kafka info that @KalmanMeth will be able to use ... something that has to do with topics, single vs. bulk messages ... etc?

Actually, on that front, I am in favor of a much more relaxed approach ... if we have a sub-set of the information and it does provide value I would not advise dropping it. We can implement those checks if you think that this is needed. 

@KalmanMeth your thoughts ???

Reply:
I think our kafka ingester should be as generic as possible, and we can add decoders to handle different types of input. If each json map is a single kafka record, I think we are fine. I am adding to the code the ability to process all the entries available in the topic, so we can process multiple lines in each loop of the process.

Reply:
@KalmanMeth agree :-) ./... ok

Reply:
I guess the format for the flows is still to be decided: JSON vs IPFIX vs any other binary encoding, so it's fine to keep it as generic as possible at the moment.